plugins/content-length.rst

```diff
@@ -1,4 +1,35 @@
 Content-Length Plugin
 =====================
 
-TODO
+The `ContentLengthPlugin` sets the correct `Content-Length` header value based on the size of the body stream of the
+request. This helps HTTP servers to handle the request.
+
+If the size of the stream can not be determined, the plugin sets the Encoding header to `chunked`, as defined in
+:rfc:`7230#section-4.1`::
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Client\Plugin\PluginClient;
+    use Http\Client\Plugin\ContentLengthPlugin;
+
+    $contentLengthPlugin = new ContentLengthPlugin();
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$contentLengthPlugin]
+    );
+
+This plugin is useful when you want to transfer data of unknown size to an HTTP application without consuming memory.
```

Comment:
It's more like, `its a unknown size until you read the whole stream`, i'm talking about chunked yes.